code/auth-service/src/permissions/permissions.js

```diff
@@ -6,42 +6,70 @@ const roleDelim = ':';
 
 const permissionAttributes = yaml.safeLoad(fs.readFileSync(config.get('permissionAttributes')));
 
+//    [ { role: 'instance-admin' }, { role: 'legacy-project:admin' } ]
+// -> [ 'instance-admin', 'project:admin' ]
+const extractInstanceRoles = entry => entry.role.replace('legacy-project', 'project');
+
+//    [ 'instance-admin', 'project:admin' ]
+// -> [ ['instance-admin'], ['project','admin'] ]
 const parseRoles = role => role.split(roleDelim, 2);
```

Comment:
Given the new data structure, I'm not sure why this function is still required. Roles will be returned from the database as first class objects won't they?